### PR TITLE
chore(cleanup): Flatten app/src -> src; delete palette folder

### PR DESCRIPTION
**Pull request by damassi**
### Description

More simplification: 
- deletes the app/src hierarchy, flattening app into src root 
- removes unnecessary aliasing
- moves the screen component into components/Screen`

cc @artsy/mobile-platform

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  It looks like you've modified a store model. If necessary, please be sure to add a migration to `migrations.ts`. See the [migration-docs](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a98e4edf082cc66bfc85bc64c01145c430245969
</p>

File: babel.config.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-const moduleResolverAlias = require("./alias").babelModuleResolverAlias
```
Comment (pvinis) [inline babel.config.js:1]: just came here looking for this, and saw its gone. is that working ok without it?

File: babel.config.js [hunk 2]
```diff
@@ -10,12 +8,18 @@ const presets = [
 ]
 
 const plugins = [
-  // the relay plugin should run before other plugins or presets
+  // The relay plugin should run before other plugins or presets
   // to ensure the graphql template literals are correctly transformed
   "relay",
-  ["module-resolver", { alias: moduleResolverAlias }],
   "import-graphql", // to enabling import syntax for .graphql and .gql files.
   "react-native-reanimated/plugin", // This has to be listed last according to the documentation. https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation/#babel-plugin
```
Comment (pvinis) [inline babel.config.js:15]: also this should totally go last, like the comment says.

File: babel.config.js (hunk 1 above)
Comment (damassi) [reply babel.config.js:1]: Yeah! works just fine without

File: babel.config.js (hunk 2 above)
Comment (damassi) [reply babel.config.js:15]: Ooh, will fix on thursday! i wonder if this might have given us some issues since the full migration... 🤔

**Pull request merged**